native/proxy-process-manager.ts

```diff
@@ -0,0 +1,108 @@
+import { spawn, ChildProcess } from "child_process";
+import { CircularBuffer } from "mnemonist";
+
+export interface ProcessResult {
+  status: number | null;
+  signal: NodeJS.Signals | null;
+  // The last 200 lines of stdout and stderr combined.
+  output: string;
+}
+
+// The `ProxyProcessManager` runs and kills the proxy process and allows us to
+// obtain the process result when the process exits.
+//
+// Stdout and stderr of the child process are forwarded to stdout and
+// stderr of this process. The lines of both these streams are also
+// written to a circular buffer with a capacity of 200 lines.
+export class ProxyProcessManager {
+  private childProcess: ChildProcess | undefined;
+  private proxyPath: string;
+  private proxyArgs: string[];
+  private enabled: boolean;
+
+  // Create a new ProxyProcessManager.
+  //
+  // `proxyPath` is the path to the executable and `proxyArgs` are the command
+  // line arguments.
+  //
+  // If `enabled` is set to false no proxy, is started and `run()` never
+  // finishes.
+  constructor(proxyPath: string, proxyArgs: string[], enabled: boolean) {
+    this.proxyPath = proxyPath;
+    this.proxyArgs = proxyArgs;
+    this.enabled = enabled;
+  }
+
+  // Run the proxy process and return `ProcessResult` when it exits.
+  //
+  // Throws an error if a process is already running.
+  async run(): Promise<ProcessResult> {
+    if (!this.enabled) {
+      return new Promise(() => undefined);
+    }
+
+    if (this.childProcess !== undefined) {
+      throw new Error("Proxy process already started");
+    }
+    const outputBuffer = new CircularBuffer<string>(Array, 200);
```

Comment:
Is there a reason why we keep it so low? Would opt for `10000`.

Reply:
`10000` might be a bit much, these logs are meant to be posted in issues and so on. Maybe somewhere around 500-1000 is managable.